pkg/utils/versions.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import (
+	"encoding/xml"
+	"fmt"
+	"github.com/Masterminds/semver/v3"
+	"github.com/hashicorp/go-retryablehttp"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+const (
+	repositoryUrl = "https://armory-cli-releases.s3.amazonaws.com/"
+
+	cliPrefix = "cli"
+)
+
+var client = retryablehttp.NewClient()
+
+// GetAllVersions returns versions available in S3 repository, sorted as SemVers in descending order.
+func GetAllVersions() ([]string, error) {
+	var versions []*semver.Version
+	more := true
+	params := url.Values{}
+	params.Set("list-type", "2")
+	params.Set("prefix", cliPrefix)
+	for more {
+		u := buildUrl(params)
+		httpRes, err := client.Get(u)
+		if err != nil {
+			return []string{}, err
+		}
+
+		var res listBucketResult
+		if err = xml.NewDecoder(httpRes.Body).Decode(&res); err != nil {
```

Comment:
```suggestion
		if err := xml.NewDecoder(httpRes.Body).Decode(&res); err != nil {
```
there is no strong reason to do this, but it always feels like polite coding to me. with the colon, the err is scoped to just the `if` block